### PR TITLE
BHBC-981: Update create project page to wizard style

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-981

## This PR contains the following changes

- Move create project components to /create folder
- Add stepper wizard component
- Remove mouse scroll zoom on leaflet map

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

**Comment by NickPhura** [comment]
I still need to write tests for the the StepperWizard component, but need to run to an appointment, so I'll have to do them later or tomorrow.  I can add them in this PR or a separate one, doesn't matter to me.

File: app/src/features/projects/create/CreateProjectPage.tsx [hunk 1]
```diff
@@ -131,24 +129,34 @@ const CreateProjectPage: React.FC = () => {
 
   const [enableCancelCheck, setEnableCancelCheck] = useState(true);
 
-  // Tracks the active step #
+  // Tracks the active  #
```
Comment (sdevalapurkar) [inline app/src/features/projects/create/CreateProjectPage.tsx]: nit:
I think step # makes sense

File: app/src/features/projects/create/CreateProjectPage.tsx [hunk 2]
```diff
@@ -458,11 +523,11 @@ const CreateProjectPage: React.FC = () => {
   /**
    * Returns the step index for the first invalid form step, or `-1` if all steps are valid
    *
-   * @return {*} {number}
+   * @return {*} {number}F
```
Comment (sdevalapurkar) [inline app/src/features/projects/create/CreateProjectPage.tsx]:
```suggestion
   * @return {*} {number}
```

**Review by sdevalapurkar — COMMENTED** [review]
looking nice :)

**Review by anissa-agahchen — COMMENTED** [review]
Found a couple of issues, no sure that they are entirely caused by your PR ... but maybe caused by validation changes:

Screen Shot 2021-05-03 at 4 45 46 PM

This one above is an issue if you try to edit the objectives section with an empty caveat.  If you touch the caveat, the issue goes away
Screen Shot 2021-05-03 at 4 48 11 PM

This 500 error occurred when I tried to save a funding source without an end-date

**Comment by sdevalapurkar** [comment]
> I still need to write tests for the the StepperWizard component, but need to run to an appointment, so I'll have to do them later or tomorrow. I can add them in this PR or a separate one, doesn't matter to me.

I think separate PR would be better 😄

**Comment by NickPhura** [comment]
> 
> 
> Found a couple of issues, no sure that they are entirely caused by your PR ... but maybe caused by validation changes:
> 

> 
> This one above is an issue if you try to edit the objectives section with an empty caveat. If you touch the caveat, the issue goes away

> 
> This 500 error occurred when I tried to save a funding source without an end-date

It looks like the issue was that the caveats field, in the API, was defaulting to `null` when no value was provided, rather than `empty string`. Updated to be empty string, which is what the objectives field was already doing.

File: app/src/components/stepper-wizard/StepperWizard.test.tsx [hunk 3]
```diff
@@ -0,0 +1,323 @@
+import TextField from '@material-ui/core/TextField';
+import { cleanup, fireEvent, render, RenderResult, waitFor } from '@testing-library/react';
```
Comment (sdevalapurkar) [inline app/src/components/stepper-wizard/StepperWizard.test.tsx]:
```suggestion
import { cleanup, fireEvent, render, waitFor } from '@testing-library/react';
```

File: app/src/components/stepper-wizard/StepperWizard.test.tsx [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+}
+
+export const SampleFormikFormYupSchema = yup.object().shape({
+  testField: yup.string().required('You must provide a value for testField')
+});
+
+const SampleFormikForm = () => {
+  const formikProps = useFormikContext<ISampleFormikFormProps>();
+
+  const { values, handleChange, handleSubmit } = formikProps;
```
Comment (sdevalapurkar) [inline app/src/components/stepper-wizard/StepperWizard.test.tsx]:
```suggestion
  const { values, handleChange } = formikProps;
```

**Review by sdevalapurkar — APPROVED** [review]
lgtm

**Comment by anissa-agahchen** [comment]
Sorry, I found one more ... the validation is preventing me from creating a project without a permit
Screen Shot 2021-05-04 at 3 52 39 PM

**Comment by sdevalapurkar** [comment]
> Sorry, I found one more ... the validation is preventing me from creating a project without a permit
> Screen Shot 2021-05-04 at 3 52 39 PM

if you press add permit then it is supposed to
if you have no permits showing in the list then does it still prevent??

**Comment by anissa-agahchen** [comment]
> > Sorry, I found one more ... the validation is preventing me from creating a project without a permit
> > Screen Shot 2021-05-04 at 3 52 39 PM
> 
> if you press add permit then it is supposed to
> if you have no permits showing in the list then does it still prevent??

Actually, it's fine .. sorry for some reason I was expecting to not need those other two fields

**Review by anissa-agahchen — APPROVED** [review]
Looks really good

**Pull request merged**